cmake/modules/zlib.cmake

```diff
@@ -70,7 +70,7 @@ else()
 				PREFIX "${PROJECT_BINARY_DIR}/zlib-prefix"
 				URL "https://github.com/madler/zlib/releases/download/v1.3.1/zlib-1.3.1.tar.gz"
 				URL_HASH "SHA256=9a93b2b7dfdac77ceba5a558a580e74667dd6fede4585b91eefb60f03b72df23"
-				CONFIGURE_COMMAND CFLAGS=${ZLIB_CFLAGS} ./configure --prefix=${ZLIB_SRC}
+				CONFIGURE_COMMAND env "CFLAGS=${ZLIB_CFLAGS}" ./configure --prefix=${ZLIB_SRC}
```

Comment:
Why `env` here?

Reply:
Question, isn't this working?
```
CONFIGURE_COMMAND CFLAGS="${ZLIB_CFLAGS}" ./configure --prefix=${ZLIB_SRC}
```

Reply:
Or, you could try `LIST_SEPARATOR` `ExternalProject_Add` option: https://cmake.org/cmake/help/latest/module/ExternalProject.html#miscellaneous-options

We use it here: https://github.com/falcosecurity/falco/blob/6adc54c92f00c7dd535323bba2bf9f02acf6eb14/cmake/modules/mimalloc.cmake#L65

Reply:
> CONFIGURE_COMMAND CFLAGS="${ZLIB_CFLAGS}" ./configure --prefix=${ZLIB_SRC}

Nope, tried that - it's the command executor that (incorrectly) splits and the inclusion of quotes here doesn't change that behavior.

> Or, you could try LIST_SEPARATOR ExternalProject_Add option: https://cmake.org/cmake/help/latest/module/ExternalProject.html#miscellaneous-options

I'll look at that, ty.

Reply:
> This doesn't work, because we aren't using cmake directly to build zlib, we're overriding/providing our own CONFIGURE_COMMAND which cmake just invokes.

Oh i see.
Since `zlib` builds with cmake, we should leverage it; are you willing to do the work?
Thanks!

Reply:
I tried that, but ran into issues with build paths that are (likely) why whoever wrote this chose this approach in the first place.

Agree we should just use cmake directly with zlib here ultimately.

I'll take a look/raise an issue, but I would prefer to merge this small fix now - since it fixes PIC builds which are currently broken, and has no other side effects.